aries/agents/mediator-client/src/mediator_client.rs

```diff
@@ -0,0 +1,23 @@
+use messages::msg_fields::protocols::out_of_band::invitation::Invitation as OOBInvitation;
+use reqwest::{self, Url};
+
+use crate::errors::MediatorClientResult;
+pub struct MediatorClient {
+    url: Url,
+    client: reqwest::Client,
+}
+
+impl MediatorClient {
```

Comment:
I am trying to envision what should the APIs here look like. Just brainstorming

```
 // mediation coordination protocol APIs https://github.com/hyperledger/aries-rfcs/blob/main/features/0211-route-coordination/README.md
  pub async fn request_mediation()
  pub async fn add_mediation_key(key: String)
  pub async fn del_mediation_key(key: String)
  pub async fn get_mediation_keys()
  
   // pickup protocol APIs https://github.com/hyperledger/aries-rfcs/tree/main/features/0685-pickup-v2
  pub async fn get_key_status(key: String)
  pub async fn get_key_messages(key: String)
```

For now we can omit things related to live delivery, our mediator itself doesn't yet support websockets.

It would be supercool to spinup acapy in mediator mode and test the client implementation against that - ultimately this should be agnostic of particular mediator implementation